src/main/java/no/unit/nva/cristin/projects/FetchOneCristinProject.java

```diff
@@ -36,12 +41,20 @@ public FetchOneCristinProject(CristinApiClient cristinApiClient, Environment env
     protected NvaProject processInput(Void input, RequestInfo requestInfo, Context context)
         throws ApiGatewayException {
 
+        validateThatSuppliedQueryParamsIsSupported(requestInfo);
+
         String language = getValidLanguage(requestInfo);
         String id = getValidId(requestInfo);
 
         return getTransformedProjectFromCristin(id, language);
     }
 
+    private void validateThatSuppliedQueryParamsIsSupported(RequestInfo requestInfo) throws BadRequestException {
+        if (!VALID_QUERY_PARAMS.containsAll(requestInfo.getQueryParameters().keySet())) {
+            throw new BadRequestException(ERROR_MESSAGE_INVALID_QUERY_PARAMS_ON_LOOKUP);
+        }
+    }
```

Comment:
This is the same code as the one above. Would it be worth to reuse the method?

Reply:
There is 2 different Set of VALID_QUERY_PARAMS. If moved to CristinHandler I need to pass both Set and Error message as well as requestInfo to the method. Do you want me to do that or keep it as is?

Reply:
leave as is